### PR TITLE
1042: Watchdog causing multiple restarts for mlbridge

### PR DESCRIPTION
**Pull request by erikj79**
When starting certain bots with a fresh scratch area, we currently end up in a restart loop. This is because all the threads immediately get busy cloning repos, which starves out the watchdog pings for longer than the hard coded 10 minutes. This patch changes the watchdog to use the configuration setting "watchdog" for the restart timeout instead. This value is currently used for a log warning which is also driven by the watchdog, so to be able to still have separate values, I've introduced a new option "watchdog_warn" which can optionally be set for just the warning part.

In addition to this, I also added a bit more logging to make it easier to follow through logstash when watchdog pings occur, or when a new instance of a bot runner is started. Failure to start due to configuration errors are now also posted using proper logs.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [SKARA-1042](https://bugs.openjdk.java.net/browse/SKARA-1042): Watchdog causing multiple restarts for mlbridge

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - Committer) ⚠️ Review applies to bb652d78e9a20946dd262d36a78cce5e8eff9854
 * [Erik Helin](https://openjdk.java.net/census#ehelin) (@edvbld - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/skara pull/1157/head:pull/1157` \
`$ git checkout pull/1157`

Update a local copy of the PR: \
`$ git checkout pull/1157` \
`$ git pull https://git.openjdk.java.net/skara pull/1157/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1157`

View PR using the GUI difftool: \
`$ git pr show -t 1157`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/skara/pull/1157.diff">https://git.openjdk.java.net/skara/pull/1157.diff</a>

</details>

### CODE REVIEW

**Review by kevinrushforth — APPROVED** [review]
Looks good.

File: bots/cli/src/main/java/org/openjdk/skara/bots/cli/BotLauncher.java [hunk 1]
```diff
@@ -163,20 +163,21 @@ public static void main(String... args) {
...
         var log = Logger.getLogger("org.openjdk.skara.bots.cli");
+        log.info("Starting BotLauncher");
 
         BotRunnerConfiguration runnerConfig = null;
         try {
             runnerConfig = BotRunnerConfiguration.parse(jsonConfig, jsonFile.getParent());
         } catch (ConfigurationError configurationError) {
-            System.out.println("Failed to parse configuration file: " + jsonFile);
-            System.out.println("Error message: " + configurationError.getMessage());
+            log.severe("Failed to parse configuration file: " + jsonFile
```
Comment (edvbld) [inline bots/cli/src/main/java/org/openjdk/skara/bots/cli/BotLauncher.java:172]: Will this actually log anything if there is no logging configured? `applyLogging` resets the `LogManager` and the `BotConsoleHandler` is only added if `log.console` is configured. Would it be wise to both print to standard out and log?

**Review by edvbld — APPROVED** [review]
Looks good. Please see my question/comment in-line.

**Review by edvbld — APPROVED** [review]
Looks good!

**Pull request closed without merging**